pkg/shortnames/shortnames.go

```diff
@@ -412,6 +431,26 @@ func ResolveLocally(ctx *types.SystemContext, name string) ([]reference.Named, e
 
 	var candidates []reference.Named
 
+	// Complete the candidates with the specified registries.  Note that
+	// "localhost/" has precedence.
+	completeCandidates := func(registries []string) ([]reference.Named, error) {
+		for _, reg := range append([]string{"localhost"}, registries...) {
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("%s/%s", reg, name))
+			if err != nil {
+				return nil, errors.Wrapf(err, "creating reference with unqualified-search registry %q", reg)
+			}
+			// Make sure to add ":latest" if needed
+			named = reference.TagNameOnly(named)
+
+			candidates = append(candidates, named)
+		}
+		return candidates, nil
+	}
+
+	if ctx != nil && ctx.ResolveShortNamesToDockerHub {
+		return completeCandidates([]string{"docker.io"})
```

Comment:
So per the earlier conversation, we should’t be including `localhost/` here, I think. That also probably implies no `completeCandidates` helper.